README.md

```diff
@@ -1,3 +1,288 @@
+{
+    [HttpPost("person")]
+    public async Task<IActionResult> CreatePerson(PersonCreated person)
+    {
+        // Publish notification (handled immediately)
+        await notificationPublisher.NotifyAsync(person);
+        return Ok();
+    }
+
+    [HttpPost("product")]
+    public async Task<IActionResult> CreateProduct(ProductCreated product)
+    {
+        // Publish message (queued for processing)
+        await messagePublisher.PublishAsync(product);
+        return Accepted();
+    }
+}
+```
+
+## Usage Patterns
+
+### Notifications vs Messages
+
+SimpleTransit supports two distinct messaging patterns:
+
+#### 1. Notifications (Fire-and-Forget)
+- **Purpose**: Immediate handling of events
+- **Interface**: `INotificationHandler<T>`
+- **Publisher**: `INotificationPublisher`
+- **Execution**: Synchronous execution of all registered handlers
+- **Use Cases**: Logging, immediate side effects, real-time updates
+
+#### 2. Messages (Queued Processing)
+- **Purpose**: Reliable, background processing
+- **Interface**: `IConsumer<T>` where `T : IMessage`
+- **Publisher**: `IMessagePublisher`
+- **Execution**: Asynchronous queue processing
+- **Use Cases**: Long-running operations, batch processing, reliable delivery
+
+## Configuration Options
+
+### Manual Registration
+
+```csharp
+builder.Services.AddSimpleTransit(options =>
+{
+    // Mark that you have notification handlers
+    options.UseNotificationHandlers();
+    
+    // Mark that you have message consumers
+    options.UseMessageConsumers();
+});
+
+// Manually register your handlers
+builder.Services.AddTransient<INotificationHandler<PersonCreated>, PersonCreatedNotificationHandler>();
+builder.Services.AddTransient<IConsumer<ProductCreated>, ProductCreatedConsumer>();
+```
+
+### Automatic Registration
+
+```csharp
+builder.Services.AddSimpleTransit(options =>
+{
+    // Register all handlers and consumers from specified assembly
+    options.RegisterServicesFromAssemblyContaining<Program>();
+    
+    // Or from a specific assembly
+    options.RegisterServicesFromAssembly(typeof(MyHandler).Assembly);
+    
+    // With optional filtering
+    options.RegisterServicesFromAssembly(
+        assembly, 
+        type => type.Namespace?.StartsWith("MyApp.Handlers") == true);
+});
+```
+
+## Advanced Scenarios
+
+### Multiple Handlers
+Multiple notification handlers can be registered for the same message type:
+
+```csharp
+public class EmailNotificationHandler : INotificationHandler<PersonCreated>
+{
+    public async Task HandleAsync(PersonCreated message, CancellationToken cancellationToken)
+    {
+        // Send email
+    }
+}
+
+public class AuditNotificationHandler : INotificationHandler<PersonCreated>
+{
+    public async Task HandleAsync(PersonCreated message, CancellationToken cancellationToken)
+    {
+        // Log to audit system
+    }
+}
+```
+
+### Error Handling
+SimpleTransit propagates exceptions from handlers to allow for proper error handling:
```

Comment:
This is true only for notifications, not for Message Consumers that act in background. Clarify this.